### ISSUE TITLE
Use install_requires to automatically install required packages if they are not present

### ISSUE THREAD

**Issue by jo-tham** [pysal/pysal#866]
Hi,

Looks like a cool package - so, thanks! Just started with it and had a small issue.

When one installs pysal with pip, it doesn't install required packages. This is very unusual in my experience with python packages - usually requirements are automatically installed. Here's an example

```
mkdir pysal && cd pysal
virtualenv venv
. venv/bin/activate
pip install pysal
python -c "import pysal"
```

Gives

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/jotham/projects/pysal/venv/lib/python2.7/site-packages/pysal/__init__.py", line 40, in <module>
    import pysal.cg
  File "/Users/jotham/projects/pysal/venv/lib/python2.7/site-packages/pysal/cg/__init__.py", line 4, in <module>
    from shapes import *
  File "/Users/jotham/projects/pysal/venv/lib/python2.7/site-packages/pysal/cg/shapes.py", line 12, in <module>
    from sphere import arcdist
  File "/Users/jotham/projects/pysal/venv/lib/python2.7/site-packages/pysal/cg/sphere.py", line 15, in <module>
    import numpy
ImportError: No module named numpy
```

Is there a reason behind this? I searched the issues in case there was a decision behind this, but didn't find anything.

I know the documentation highlights to use different python distributions, or to make sure numpy and scipy are installed. but we can solve this automatically by using `install_requires` instead of `requires` [here in setup.py](https://github.com/pysal/pysal/blob/master/setup.py#L77), described [here](http://python-packaging.readthedocs.io/en/latest/dependencies.html), for example.

I think it would make it a bit easier to get started with pysal. I'm glad to open a PR.
- Platform information:

``` python
>>> import os; print(os.name, os.sys.platform);print(os.uname())
('posix', 'darwin')
('Darwin', 'Jothams-MacBook-Pro.local', '15.6.0', 'Darwin Kernel Version 15.6.0: Thu Jun 23 18:25:34 PDT 2016; root:xnu-3248.60.10~1/RELEASE_X86_64', 'x86_64')
```
- Python version: 

``` python
>>> import sys; print(sys.version)
2.7.11 (default, Jan 22 2016, 08:29:18)
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
```

**Comment by sjsrey** [comment]
Thanks for the issue. 

We would certainly be interested in a PR that addresses this pain point.

**Comment by ljwolf** [comment]
resolved by #871
